app/src/main/java/com/example/habitsmasher/FollowUserDialog.java

```diff
@@ -0,0 +1,160 @@
+                                    if (document.get(USERNAME_FIELD).toString().equals(usernameInput)){
+                                        Log.d(TAG, "Username exists");
+
+                                        // if username exists, update follower/following counts
+                                        String currentUserId = getCurrentUserId();
+
+                                        User userToFollow = document.toObject(User.class);
+                                        userToFollow.addNewFollower(currentUserId);
+
+                                        // add new follower to the user to follow
+                                        addNewFollowerForUserInDatabase(userToFollow.getId(),
+                                                                        currentUserId);
+
+                                        // add to new followed user to following of current user
+                                        addUserToFollowingForUserInDatabase(currentUserId,
+                                                                            userToFollow.getId());
+
+                                        showFollowSuccessMessage();
+
+                                        followUserDialog.dismiss();
+
+                                        return;
+                                    }
+                                }
+                            }
+                            // if username does not exist, throw error message on the front-end
+                            Log.d(TAG, "Username does not exist");
+
+                            displayErrorMessage(INVALID_USERNAME_ERROR);
+                        } else {
+                            Log.d(TAG, "Failed with: ", task.getException());
+                        }
+                    }
+                });
+            }
+        });
+        return view;
+    }
+
+    private void showFollowSuccessMessage() {
+        Toast.makeText(getContext(),
+                       USER_FOLLOWED_SUCCESS_MESSAGE,
+                       Toast.LENGTH_LONG)
+             .show();
+    }
+
+    private void addUserToFollowingForUserInDatabase(String userId, String followedUserId) {
+        DocumentReference userRef = _db.collection(USERS_COLLECTION_PATH).document(userId);
+        userRef.update(FOLLOWING_FIELD, FieldValue.arrayUnion(followedUserId));
+    }
+
+    private void addNewFollowerForUserInDatabase(String userId, String newFollowerId) {
+        DocumentReference userRef = _db.collection(USERS_COLLECTION_PATH).document(userId);
+        userRef.update(FOLLOWERS_FIELD, FieldValue.arrayUnion(newFollowerId));
+    }
+
+
+    public String getCurrentUserId() {
+        SharedPreferences sharedPref = getContext().getSharedPreferences(USER_DATA_PREFERENCES_TAG, Context.MODE_PRIVATE);
+        return sharedPref.getString(USER_ID_SHARED_PREF_TAG, "id");
+    }
+
+    @Override
+    public void displayErrorMessage(int messageType) {
```

Comment:
Following the implementation by @jaskim9824 when overriding the methods in the `DisplaysErrorMessages` interface